### PR TITLE
calculate percentage of balanced strings

### PR DESCRIPTION
**Pull request by iliamamageishvili**
## What's the problem?

Call the service N = 1000 times for each of the strings of the length 2,4,8.
Calculate the percent of balanced strings for each length and print the results to stdout.

## How did you solve it?

 i set up an HTTP server that listens for requests to generate a sequence, and spawns a Goroutine to handle the server. It then tests the generated sequences for balance and reports the percentage of balanced sequences for different lengths. Finally, it adds a sleep to ensure that the Goroutine has enough time to start before the program ends.

## How can this be tested?

By running main

## Checklist

- [x] I have tested my changes.
- [x] I have updated the documentation if necessary.
- [x] My code follows the project's code style guidelines.
- [x] My code has been reviewed by at least one other team member.

### CODE REVIEW

**Review by xartishuk — CHANGES_REQUESTED** [review]
After this change you will not be able to run a server anymore! Keep the server and percentage calculator separate. For that create 2 mains: `cmd/server/main.go` and `cmd/stats/main.go`.

And let's make your stats program into a CLI, with length (`2`,`4`,`8`), number of iterations (`1000`) and server host (`http://localhost:8080`) turned into arguments (hint: use https://pkg.go.dev/flag).

P.S. for length you don't need to accept all three different length at the same time. You can make it so your CLI only accepts one length at a time, and require 3 separate calls to achieve what the task requires

File: cmd/stats/main.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	validLength := false
+
+	for _, l := range validLengths {
+		if *length == l {
+			validLength = true
+			break
+		}
+	}
+	if !validLength {
+		fmt.Println("Invalid length, must be one of 2, 4, or 8")
```
Comment (xartishuk) [inline cmd/stats/main.go]: don't check 2, 4, 8; just check that it's positive

File: cmd/stats/main.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		return
+	}
+
+	if *iterations <= 0 {
+		fmt.Println("Invalid number of iterations, must be positive")
+		return
+	}
+
+	rand.Seed(time.Now().Unix())
```
Comment (xartishuk) [inline cmd/stats/main.go]: why do you need this?

**Review by xartishuk — CHANGES_REQUESTED** [review]

File: cmd/stats/main.go (hunk 2 above)
Comment (iliamamageishvili) [reply cmd/stats/main.go]: I missed to remove it its from previous version and is not relevant at this code

**Review by xartishuk — APPROVED** [review]

**Pull request merged**